### PR TITLE
Upgrade the nix+nixpkgs versions

### PR DESCRIPTION
**Pull request by jwiegley**

### CODE REVIEW

File: hnix.cabal [hunk 1]
```diff
@@ -516,7 +872,7 @@ library
     , monadlist >= 0.0.2 && < 0.1
     , mtl >= 2.2.2 && < 2.3
     , optparse-applicative >= 0.14.3 && < 0.15 || >= 0.15.0.0 && < 0.16
-    , parser-combinators >= 1.0.1 && < 1.1 || >= 1.2.0 && < 1.3
+    , parser-combinators
```
Comment (mightybyte) [inline hnix.cabal:875]: Bump the upper bound here instead of dropping them entirely?

Comment (jwiegley) [reply hnix.cabal:875]: Yeah, better idea that.

**Comment by sjakobi** [comment]
@jwiegley Does this fix the build issue you mentioned in https://github.com/haskell-nix/hnix/issues/524#issuecomment-554208221?

Since you added the "help wanted"-label: How can I or others help here?

**Comment by jwiegley** [comment]
If anyone can update the Nix submodule, update the version of Nix using in `default.nix`, and update the `nixpkgs` pin, and see it build on Darwin and Linux, then we'll merge it.

**Pull request closed without merging**